app/src/test/java/com/codesoom/assignment/product/domain/ProductRepositoryTest.java

```diff
@@ -0,0 +1,151 @@
+    class Describe_findById {
+        @Nested
+        @DisplayName("유효한 ID가 주어지면")
+        class Context_with_existed_id extends JpaTest {
+            private Product givenProduct;
+
+            @BeforeEach
+            void prepare() {
+                givenProduct = getProductRepository().save(ProductFactory.createProduct());
+            }
+
+            @Test
+            @DisplayName("고양이 장난감을 찾아서 리턴한다")
+            void it_returns_searched_cat_toy() {
+                final Product actualProduct = getProductRepository().findById(givenProduct.getId()).orElse(null);
+
+                assertThat(actualProduct.getName()).isEqualTo(givenProduct.getName());
+                assertThat(actualProduct.getId()).isEqualTo(givenProduct.getId());
+                assertThat(actualProduct.getPrice()).isEqualTo(givenProduct.getPrice());
+            }
+        }
+
+        @Nested
+        @DisplayName("유효하지 않은 ID가 주어지면")
+        class Context_with_non_existed_id extends JpaTest {
+            @Test
+            @DisplayName("Optional.empty()를 리턴한다")
+            void it_returns_optional_empty() {
+                assertThat(getProductRepository().findById(100L)).isEmpty();
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("save 메소드는")
+    class Describe_save {
+        @Nested
+        @DisplayName("새로운 고양이 장난감이 주어지면")
+        class Context_with_new_cat_toy extends JpaTest {
+            private final Product givenProduct = ProductFactory.createProduct();
+
+            @Test
+            @DisplayName("DB에 등록하고 등록된 장난감을 리턴한다")
+            void it_returns_registered_cat_toy() {
+                final Product actualProduct = getProductRepository().save(givenProduct);
+
+                assertThat(actualProduct.getName()).isEqualTo(givenProduct.getName());
+                assertThat(actualProduct.getPrice()).isEqualTo(givenProduct.getPrice());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("delete 메소드는")
+    class Describe_delete {
+        @Nested
+        @DisplayName("유효한 ID가 주어지면")
+        class Context_with_existed_id extends JpaTest {
+            private Product givenProduct;
+
+            @BeforeEach
+            void prepare() {
+                givenProduct = getProductRepository().save(ProductFactory.createProduct());
+            }
+
+            @Test
+            @DisplayName("DB에서 삭제한다")
+            void it_returns_nothing() {
+                getProductRepository().delete(givenProduct);
+
+                final Optional<Product> actualProduct = getProductRepository().findById(givenProduct.getId());
+
+                assertThat(actualProduct).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("유효하지 않은 ID가 주어지면")
+        class Context_with_non_existed_id extends JpaTest {
+            private final Product givenProduct = ProductFactory.createProduct(100L);
+
+            @Test
+            @DisplayName("삭제할 대상이 없기때문에 아무것도 삭제하지않는다")
```

Comment:
테스트 코드에 대해 원인을 분석하는 설명이 들어가지 않는 것이 좋습니다. 구현이 얼마든지 바뀔 수 있기 때문이죠.
```suggestion
            @DisplayName("삭제하지않는다")
```